lib/tasks/reports.rake

```diff
@@ -0,0 +1,15 @@
+namespace :reports do
+  desc "Gives report for concatenated price rules"
+  task concatenated_price_rules: :environment do
+    policies = {}
+    PricePolicy.current.group_by { |p| [p.product_id, p.price_group_id] }.map { |k, ps| policies[k] = ps if ps.count > 1 }
```

Comment:
I'd go with `each` over `map` here since you're not directly capturing its result. Either that or consider refactoring using `each_with_object`.